app/Tools/Database.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Tools;
+
+use Exception;
+use Illuminate\Support\Str;
+use PDO;
+
+class Database
+{
+    private $dsn;
+    private $username;
+    private $password;
+
+    public function url(string $url)
+    {
+        $url = parse_url($url);
+        $databaseType = $url['scheme'];
+        $host = $url['host'];
+        $port = $url['port'];
+        $this->username = $url['user'];
+        $this->password = $url['pass'];
+
+        $this->dsn = "{$databaseType}:host={$host};port={$port}";
+        return $this;
+    }
+
+    public function find(string $schema = null): bool
```

Comment:
can we replace `$schema` with `$databaseName`?

Reply:
Also, is this method more like a `exists`?

Reply:
Done.